pkg/stream/vcenter.go

```diff
@@ -0,0 +1,76 @@
+package stream
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/pkg/errors"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/connection"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/processor"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/event"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	authMethodvSphere = "user_password"
+)
+
+// vCenter handles the connection to vCenter to retrieve an event stream
+type vCenter struct {
+	client *govmomi.Client
+	stream *event.Manager
+}
+
+// NewVCenterStream returns a vCenter event manager
+func NewVCenterStream(ctx context.Context, cfg connection.Config) (Streamer, error) {
+	var v vCenter
+	u, err := soap.ParseURL(cfg.Address)
+	if err != nil {
+		return nil, errors.Wrap(err, "error parsing URL")
+	}
+
+	var username, password string
+	switch cfg.Auth.Method {
+	case authMethodvSphere:
+		username = cfg.Auth.Secret["username"]
+		password = cfg.Auth.Secret["password"]
+	default:
+		return nil, errors.Errorf("unsupported authentication method for source vCenter: %s", cfg.Auth.Method)
+	}
+	u.User = url.UserPassword(username, password)
+
+	var insecure bool
+	if cfg.Options["insecure"] == "true" {
+		insecure = true
+	}
+
+	client, err := govmomi.NewClient(ctx, u, insecure)
+	if err != nil {
+		return nil, err
+	}
+
+	v.client = client
+	v.stream = event.NewManager(client.Client)
+	return &v, nil
+}
+
+// Stream is the main logic, blocking to receive and handle events from vCenter
+func (v *vCenter) Stream(ctx context.Context, p processor.Processor) error {
```

Comment:
As per first comment `v` -> `vCenter` if struct is renamed `VCenter`

Reply:
I think parameters should be single words. But if `vCenter` is already taken, I wonder if `vCtr` would suffice.

Reply:
changed to `vcenter`